modules/calendar/add.php

```diff
@@ -0,0 +1,354 @@
+<?php
+require_once("../../interface/globals.php");
+require_once("$srcdir/headers.inc.php");
+require_once("$srcdir/htmlspecialchars.inc.php");
+require_once("$srcdir/patient.inc");
+require_once("$srcdir/formatting.inc.php");
+require_once("$srcdir/calendar.inc");
+require_once("$srcdir/encounter_events.inc.php");
+
+$dateFormat = DateFormatRead();
+call_required_libraries(array("jquery-min-3-1-1","bootstrap","datepicker"));
+
+// create event slots according to:
+
+// 1. Provider Vacation dates
+if ($_POST['form_action'] == "vacation_submit") {
+  $event_date = $_POST['startDate'];  // Vacation starts at this date
+  $event_end_date = $_POST['endDate'];  // Vacation ends at this date
+  $starttime = "{$GLOBALS['schedule_start']}:00:00";
+  $endtime = "{$GLOBALS['schedule_end']}:00:00";
+  $provider = $_POST['provider_id'];
+
+  // Event location specifications
+  $locationspecs = array( "event_location" => "",
+                          "event_street1" => "",
+                          "event_street2" => "",
+                          "event_city" => "",
+                          "event_state" => "",
+                          "event_postal" => "" );
+  $locationspec = serialize($locationspecs);
+
+  // Recurring event specifications
+  $recurrspec = array( "event_repeat_freq" => "1", // every
+                       "event_repeat_freq_type" => "0",  // day
+                       "event_repeat_on_num" => "1",
+                       "event_repeat_on_day" => "0",
+                       "event_repeat_on_freq" => "0",
+                       "exdate" => "" );
+  $args = array();
+    // specify values needed for the INSERT into table - libreehr_postcalendar_events - columns
+    $args['form_category'] = "4";  // Vacation
+    $args['new_multiple_value'] = "";
+    $args['form_provider'] = $provider;  // selected provider
+    $args['form_title'] = "Vacation";
+    $args['form_comments'] = "";
+    $args['event_date'] = $event_date;
+    $args['form_enddate'] = $event_end_date;
+    $args['duration'] = 0;  // no duration in provider events
+    $args['form_repeat'] = "1";  // Repeats box checked
+    $args['recurrspec'] = $recurrspec;
+    $args['starttime'] = $starttime;
+    $args['endtime'] = $endtime;
+    $args['form_allday'] = 0;
+    $args['form_apptstatus'] = "-";  // None
+    $args['form_prefcat'] = "0";  // None
+    $args['locationspec'] = $locationspec;
+
+    $def_facility = sqlFetchArray(sqlStatement( "SELECT u.facility_id
+                                                 FROM users u
+                                                 LEFT JOIN facility f on (u.facility_id = f.id)
+                                                 WHERE u.id = ?", array($_SESSION['authId']) ));
+    // default_user_facility denotes id of default facility of logged in user
+    $default_user_facility = $def_facility['facility_id'];
+    // Taking facility & billing facility as default facility of logged in user
```

Comment:
Since provider is on a vacation, I assumed this. Is this assumption fine?

Reply:
Yes that is fine . They should only be working with the facility that they have access to.